build.sbt

```diff
@@ -25,3 +25,28 @@ lazy val core = (projectMatrix in file("core"))
     libraryDependencies ++= Seq(Libraries.uPickle) ++ Libraries.sttpClient ++ Seq(Libraries.scalaTest)
   )
   .settings(commonSettings: _*)
+
+//TODO this should be invoked by compilation process, see #https://github.com/scalameta/mdoc/issues/355
+val compileDocs: TaskKey[Unit] = taskKey[Unit]("Compiles docs module throwing away its output")
+compileDocs := {
+  (docs.jvm(scala2.head) / mdoc).toTask(" --out target/sttp-openai-docs").value
+}
+
+lazy val docs = (projectMatrix in file("generated-docs")) // important: it must not be docs/
+  .enablePlugins(MdocPlugin)
+  .settings(commonSettings)
+  .settings(
+    mdocIn := file("README.md"),
+    moduleName := "sttp-openai-docs",
+    mdocOut := file("generated-docs/out"),
+    mdocExtraArguments := Seq("--clean-target"),
+    publishArtifact := false,
+    name := "docs",
+    libraryDependencies ++= Seq(
+      "com.softwaremill.sttp.client4" %% "cats" % "4.0.0-M1",
+      "org.typelevel" %% "cats-effect" % "3.6-1f95fd7"
```

Comment:
3.6? I think 3.5.0 is the latest stable :) but sttp-client4-cats shoudl bring in the necessary dep

Reply:
Is this dependecy required, for mdoc? 
```
"org.typelevel" %% "cats-effect" % "3.6-1f95fd7"
```

Reply:
Unfortunately, it doesn't
```scala
error: README.md:93:1: object ExitCode is not a member of package cats.effect
import cats.effect.{ExitCode, IO, IOApp}
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
error: README.md:101:21: not found: type IOApp
object Main extends IOApp {
                    ^^^^^
error: README.md:102:41: not found: type IO
  override def run(args: List[String]): IO[ExitCode] = {
                                        ^^
error: README.md:116:36: not found: type IO
    HttpClientCatsBackend.resource[IO]().use { backend =>
                                   ^^
```
and more. But you are right, should have used latest stable `3.5.0` :)

Reply:
Ah, the dependency in sttp is only on the kernel. So you'll need it, yeah